app/models/partner.rb

```diff
@@ -247,6 +247,47 @@ def has_service_areas?
     service_areas.any?
   end
 
+  def can_clear_address?(user = nil)
```

Comment:
I think this is unnecessary - if you can edit a partner at all, you can clear the address - so I would just remove this check. Also you've already defined access over in the policy.

Reply:
realise I've just contradicted myself here above - the check that needs to be in here is, instead of just seeing if you're a neighbourhood admin or a partnership admin, checking to see if you admin for the neighbourhood used in the address or are a partner admin or root - these are the only people who should be able to remove an address

Reply:
okay so i kept all the logic as before but extended the checks so if you are a neighbourhood admin then you must have the address neighbourhood in your neighbourhood pool if you want to remove the address.

i tried putting in the same check for partnership admins but you can't give a partnership admin a neighbourhood without making them a neighbourhood admin, so it felt moot.

Reply:
This doesn't quite work - I'll try to explain why!

The first 5 statements are all working fine.

Then we get to  two lines you should take out:

`return false unless user.neighbourhood_admin? || user.partnership_admin?`
and
`return true if user.partnership_admin?`

If you're able to see this screen and you don't own a partner and aren't a root, you're one of these, and even if you are, just being a neighbourhood_admin shouldn't give you access to any partner - just access to partners in your neighbourhood. We only want to return true if you admin for the address - you shouldn't be allowed to remove addresses other partnership or neighbourhood admins have added, because that will eject this partner from their neighbourhood. Does that make sense?

(Would probably also remove `return false if user_hood_ids.empty?` because the following line will immediately return false if it is indeed empty, and it will make it tidier)